spring-integration-zookeeper/src/main/java/org/springframework/integration/zookeeper/metadata/ZookeeperMetadataStore.java

```diff
@@ -124,14 +122,7 @@ public String putIfAbsent(String key, String value) {
 			}
 			catch (@SuppressWarnings(UNUSED) KeeperException.NodeExistsException e) {
 				// so the data actually exists, we can read it
-				try {
-					byte[] bytes = this.client.getData().forPath(getPath(key));
-					return IntegrationUtils.bytesToString(bytes, this.encoding);
-				}
-				catch (Exception exceptionDuringGet) {
-					throw new ZookeeperMetadataStoreException("Exception while reading node with key '" + key + "':",
-							exceptionDuringGet);
-				}
+				return get(key);
```

Comment:
Gary, is this fix OK with you?
Because we would need to back-port it anyway.

Reply:
Yes, it looks fine; I'll backport it along with the other bugs in ZK from my Sonar fixes.